kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/CommitInfo.java

```diff
@@ -86,27 +115,83 @@ public static CommitInfo fromColumnVector(ColumnVector vector, int rowId) {
             : VectorUtils.toJavaMap(children[7].getMap(rowId)));
   }
 
-  public static StructType FULL_SCHEMA =
-      new StructType()
-          .add("inCommitTimestamp", LongType.LONG, true /* nullable */)
-          .add("timestamp", LongType.LONG)
-          .add("engineInfo", StringType.STRING)
-          .add("operation", StringType.STRING)
-          .add(
-              "operationParameters",
-              new MapType(StringType.STRING, StringType.STRING, true /* nullable */))
-          .add("isBlindAppend", BooleanType.BOOLEAN, true /* nullable */)
-          .add("txnId", StringType.STRING)
-          .add(
-              "operationMetrics",
-              new MapType(StringType.STRING, StringType.STRING, true /* nullable */));
+  /**
+   * Returns the `inCommitTimestamp` of delta file at the requested version. Throws an exception if
+   * the delta file does not exist or does not have a commitInfo action or if the commitInfo action
+   * contains an empty `inCommitTimestamp`.
+   */
+  // TODO: [delta-io/delta#5147] Can't just use the logPath & version on catalogManaged tables.
+  public static long getRequiredInCommitTimestampFromFile(
+      Engine engine, Path logPath, long version) {
+    return getRequiredInCommitTimestamp(
+        getCommitInfoOpt(engine, logPath, version), version, logPath);
+  }
 
-  private static final Map<String, Integer> COL_NAME_TO_ORDINAL =
-      IntStream.range(0, FULL_SCHEMA.length())
-          .boxed()
-          .collect(toMap(i -> FULL_SCHEMA.at(i).getName(), i -> i));
+  /**
+   * Returns the `inCommitTimestamp` of the given `commitInfoOpt` if it is defined. Throws an
+   * exception if `commitInfoOpt` is empty or contains an empty `inCommitTimestamp`.
+   */
+  // TODO: [delta-io/delta#5147] Can't just use the logPath & version on catalogManaged tables.
+  public static long getRequiredInCommitTimestamp(
+      Optional<CommitInfo> commitInfoOpt, long version, Path dataPath) {
+    CommitInfo commitInfo =
+        commitInfoOpt.orElseThrow(
+            () -> DeltaErrors.tableWithIctMissingCommitInfo(dataPath.toString(), version));
+    return commitInfo.inCommitTimestamp.orElseThrow(
+        () -> DeltaErrors.tableWithIctMissingIct(dataPath.toString(), version));
+  }
 
-  private static final Logger logger = LoggerFactory.getLogger(CommitInfo.class);
+  /**
+   * Get the CommitInfo action (if available) from the delta file at the given logPath and version.
+   */
+  // TODO: [delta-io/delta#5147] Can't just use the logPath & version on catalogManaged tables.
+  public static Optional<CommitInfo> getCommitInfoOpt(Engine engine, Path logPath, long version) {
```

Comment:
is this used anywhere besides `getRequiredInCommitTimestampFromFile`? I would almost prefer we just get rid of this? or at least similarly re-name this to be clear this uses the log-path delta files only